weblate/glossary/models.py

```diff
@@ -154,6 +154,11 @@ def fetch_glossary_terms(  # noqa: C901
             # Variant is used for variant grouping below, source unit for flags
             base_units = base_units.select_related("source_unit", "variant")
 
+            # Exclude fuzzy entries whose target exactly equals source
+            base_units = base_units.exclude(
+                Q(state=STATE_FUZZY) & Q(source=F("target"))
```

Comment:
What is wrong on target matching source? It can happen in many situations and it is still useful to know that the translation is not supposed to be changed.

Reply:
It's not about "target matching source".

It's about "target matching source" PLUS "target is not translated yet" (fuzzy)